src/main/java/agave/TaskList.java

```diff
@@ -0,0 +1,124 @@
+package agave;
+
+import java.util.ArrayList;
+
+/**
+ * The TaskList class manages a list of tasks, allowing tasks to be added, removed, and manipulated.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs an empty TaskList.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+
+    /**
+     * Constructs a TaskList with the specified list of tasks.
+     *
+     * @param tasks The list of tasks to be managed.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns the list of tasks.
+     *
+     * @return The list of tasks.
+     */
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param taskNumber The index of the task to be deleted.
+     * @throws AgaveException If the task number is out of range.
+     */
+    public void deleteTask(int taskNumber) throws AgaveException {
+        if (taskNumber > 0 && taskNumber <= tasks.size()) {
+            Task removed = tasks.remove(taskNumber - 1);
+            System.out.println("Removed: " + removed);
```

Comment:
Should this method be extracted into Ui? It is possible that issues arise when the output target has to be changed.